Jellyfin.Data/Entities/Security/Device.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Globalization;
+
+namespace Jellyfin.Data.Entities.Security
+{
+    /// <summary>
+    /// An entity representing a device.
+    /// </summary>
+    public class Device
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Device"/> class.
+        /// </summary>
+        /// <param name="userId">The user id.</param>
+        /// <param name="appName">The app name.</param>
+        /// <param name="appVersion">The app version.</param>
+        /// <param name="deviceName">The device name.</param>
+        /// <param name="deviceId">The device id.</param>
+        public Device(Guid userId, string appName, string appVersion, string deviceName, string deviceId)
+        {
+            UserId = userId;
+            AppName = appName;
+            AppVersion = appVersion;
+            DeviceName = deviceName;
+            DeviceId = deviceId;
+
+            AccessToken = Guid.NewGuid().ToString("N", CultureInfo.InvariantCulture);
+            DateCreated = DateTime.UtcNow;
+            DateLastActivity = DateCreated;
+
+            // Non-nullable for EF Core, as this is a required relationship.
+            User = null!;
+        }
+
+        /// <summary>
+        /// Gets the id.
+        /// </summary>
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int Id { get; private set; }
+
+        /// <summary>
+        /// Gets the user id.
+        /// </summary>
+        public Guid UserId { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the access token.
+        /// </summary>
+        public string AccessToken { get; set; }
+
+        /// <summary>
+        /// Gets or sets the app name.
+        /// </summary>
+        [MaxLength(64)]
+        [StringLength(64)]
+        public string AppName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the app version.
+        /// </summary>
+        [MaxLength(32)]
+        [StringLength(32)]
+        public string AppVersion { get; set; }
+
+        /// <summary>
+        /// Gets or sets the device name.
+        /// </summary>
+        [MaxLength(64)]
+        [StringLength(64)]
+        public string DeviceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the device id.
+        /// </summary>
+        [MaxLength(256)]
+        [StringLength(256)]
+        public string DeviceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this device is active.
+        /// </summary>
+        public bool IsActive { get; set; }
+
+        /// <summary>
+        /// Gets or sets the date this device was created.
+        /// </summary>
+        public DateTime DateCreated { get; set; }
```

Comment:
Should this be subclassed for consistency? Call it `IAuditableEntity` or whatever.

Reply:
@barronpm I was more thinking an abstract class (without a separate table), but my experience with EF is somewhat limited.

Reply:
That can be done, but what would the benefit be? If the goal is to automatically update the DateModified field, that would be done through EF Core events, or modifying the SaveChanges method, which I plan to do in a PR that moves other entities over to this interface when it's finalized.

Reply:
I don't know. That's why I phrased it as a question.

Do we want this "audit" data for every (or many) table? We can simply define them as [shadow properties](https://docs.microsoft.com/en-us/ef/core/modeling/shadow-properties) if we want and modify SaveChanges as you say.

Reply:
@barronpm if we don't have any immediate (and useful) plans for `IAuditableEntity` I think we should just remove it for now

Reply:
I agree, for now it can be pushed off as a "future" improvement, since it would be something we want standardized across entities